ios/browser/api/ledger/brave_ledger.mm

```diff
@@ -0,0 +1,2128 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#import "brave_ledger.h"
+#import <UIKit/UIKit.h>
+
+#include "base/base64.h"
+#include "base/containers/flat_map.h"
+#include "base/ios/ios_util.h"
+#include "base/logging.h"
+#include "base/sequenced_task_runner.h"
+#include "base/strings/sys_string_conversions.h"
+#include "base/task/post_task.h"
+#include "base/task/thread_pool.h"
+#include "base/task_runner_util.h"
+#include "base/time/time.h"
+#include "brave/build/ios/mojom/cpp_transformations.h"
+#import "brave/ios/browser/api/common/common_operations.h"
+#import "brave/ios/browser/api/ledger/brave_ledger_observer.h"
+#import "brave/ios/browser/api/ledger/ledger.mojom.objc+private.h"
+#import "brave/ios/browser/api/ledger/ledger_client_bridge.h"
+#import "brave/ios/browser/api/ledger/ledger_client_ios.h"
+#import "brave/ios/browser/api/ledger/legacy_database/data_controller.h"
+#import "brave/ios/browser/api/ledger/legacy_database/legacy_ledger_database.h"
+#import "brave/ios/browser/api/ledger/promotion_solution.h"
+#import "brave/ios/browser/api/ledger/rewards_notification.h"
+#include "brave/vendor/bat-native-ledger/include/bat/ledger/global_constants.h"
+#include "brave/vendor/bat-native-ledger/include/bat/ledger/ledger.h"
+#include "brave/vendor/bat-native-ledger/include/bat/ledger/ledger_database.h"
+#include "brave/vendor/bat-native-ledger/include/bat/ledger/option_keys.h"
+#include "components/os_crypt/os_crypt.h"
+#include "net/base/registry_controlled_domains/registry_controlled_domain.h"
```

Comment:
it seems like we should add `//net` to the DEPS file, wondering whether `checkdeps` says anything

Reply:
I just ran a `gn_check` and it came back all good, so I'm not 100% certain why it doesn't complain.